apps/files/src/components/AllFilesList.vue

```diff
@@ -0,0 +1,192 @@
+              <template v-if="activeFilesCount.folders > 0">
+                {{ activeFilesCount.folders }}
+                <translate :translate-n="activeFilesCount.folders" translate-plural="folders">folder</translate>
+              </template>
+              <translate v-if="activeFilesCount.folders > 0 && activeFilesCount.files > 0">and</translate>
+              <template v-if="activeFilesCount.files > 0">
+                {{ activeFilesCount.files }}
+                <translate :translate-n="activeFilesCount.files" translate-plural="files">file</translate>
+              </template>
+              <template v-if="activeFiles.length > 0">
+                &dash; {{ filesTotalSize | fileSize }}
+              </template>
+            </div>
+            <div v-if="!publicPage()" class="uk-visible@s uk-width-2-3 uk-width-1-2@xl uk-text-meta uk-flex" :class="{ 'uk-visible@xl' : _sidebarOpen  }">
+              <span class="uk-margin-small-right">
+                <translate>Used space:</translate> {{ quota.used | fileSize }}
+                <template v-if="quota.definition !== 'default' && quota.definition !== 'none'">
+                  <translate>of</translate> {{ quota.definition }}
+                </template>
+              </span>
+              <div class="uk-width-expand oc-align-self-center" v-if="quota.definition !== 'default' && quota.definition !== 'none'">
+                <oc-progress :value="parseInt(quota.relative)" :max="100" class="uk-margin-remove" />
+              </div>
+            </div>
+          </template>
+        </file-list>
+      </div>
+    </div>
+    <oc-dialog-prompt name="change-file-dialog" :oc-active="changeFileName" v-model="newName" :ocError="changeFileErrorMessage"
+                      :ocTitle="_renameDialogTitle" ocConfirmId="oc-dialog-rename-confirm" @oc-confirm="changeName"
+                      @oc-cancel="changeFileName = false; newName = ''" />
+    <oc-dialog-prompt name="delete-file-confirmation-dialog" :oc-active="filesDeleteMessage !== ''"
+                      :oc-content="filesDeleteMessage" :oc-has-input="false" :ocTitle="_deleteDialogTitle"
+                      ocConfirmId="oc-dialog-delete-confirm" @oc-confirm="reallyDeleteFiles"
+                      @oc-cancel="setFilesDeleteMessage('')"/>
+  </div>
+</template>
+<script>
+import FileList from './FileList.vue'
+import OcDialogPrompt from './ocDialogPrompt.vue'
+import { mapGetters, mapActions, mapState } from 'vuex'
+
+import Mixins from '../mixins'
+
+export default {
+  components: {
+    FileList,
+    OcDialogPrompt
+  },
+  mixins: [
+    Mixins
+  ],
+  name: 'AllFilesList',
+  props: ['fileData', 'starsEnabled', 'checkboxEnabled', 'dateEnabled', 'parentFolder'],
+  mounted () {
+    this.$_ocFilesFolder_getFolder()
+  },
+  methods: {
+    ...mapActions('Files', ['loadFolder', 'setFilterTerm', 'markFavorite',
```

Comment:
If we want this to be reusable component we shall not use any mapped actions or states or anything else related to vuex.

Any state interaction has to be done via props, event and/or maybe mixins or method inheritance.

Just saying ... ;-)

Reply:
Lets Chat tomorrow about this. Lets Team up